HW1/P6/P6.py

```diff
@@ -0,0 +1,86 @@
+############################################
+### Problem 6 - Markov Shakespeare[25%]	 ###
+### P6.py								 ###
+### Patrick Day 						 ###
+### CS 205 HW1                           ###
+### Oct 4th, 2015						 ###
+############################################
+
+########################
+### Import Functions ###
+########################
+import pyspark
+import numpy as np
+from operator import add
+
+import os
+sc = pyspark.SparkContext()
+sc.setLogLevel('WARN')
+
+#################
+### Build RDD ###
+#################
+# Import Data and Split into Words
+shake = sc.textFile('pg100.txt', use_unicode=False)
+words = shake.flatMap(lambda x: x.split())
+
+# Filter our spaces, uppercase, and digits
+filter_words = words.filter(lambda x: x != " " and x != x.upper() and x != x.isdigit())
+
+# Create RDD w/ index for "text steram" and adjecent words
+text_stream_ind = filter_words.zipWithIndex()
+txt_ind_1 = text_stream_ind.map(lambda x: (x[1], x[0]))
+
+# Create the "shifted by 1" lists
+shift_1 = filter_words.collect()[1:] 
+shift_2 = shift_1[1:]
+
+# Create RDDs for shifts
+shift_1_rdd = sc.parallelize(shift_1, 10)
+shift_2_rdd = sc.parallelize(shift_2, 10)
```

Comment:
Instead of shifting them in the lists, you could also shift them in the RDDs and then join them